client/shaders/nodes_shader/opengl_vertex.glsl

```diff
@@ -75,12 +110,19 @@ float disp_z;
 		smoothTriangleWave(animationTimer * 13.0 + tOffset)) * 0.5;
 #endif
 
+	worldPosition = (mWorld * gl_Vertex).xyz;
 
 #if (MATERIAL_TYPE == TILE_MATERIAL_WAVING_LIQUID_TRANSPARENT || MATERIAL_TYPE == TILE_MATERIAL_WAVING_LIQUID_OPAQUE || MATERIAL_TYPE == TILE_MATERIAL_WAVING_LIQUID_BASIC) && ENABLE_WAVING_WATER
+	// Generate waves with Perlin-type noise.
+	// The constants are calibrated such that they roughly
+	// correspond to the old sine waves.
 	vec4 pos = gl_Vertex;
-	pos.y -= 2.0;
-	float posYbuf = (pos.z / WATER_WAVE_LENGTH + animationTimer * WATER_WAVE_SPEED * WATER_WAVE_LENGTH);
-	pos.y -= sin(posYbuf) * WATER_WAVE_HEIGHT + sin(posYbuf / 7.0) * WATER_WAVE_HEIGHT;
+	vec3 wavePos = worldPosition + cameraOffset;
+	wavePos.x /= WATER_WAVE_LENGTH * 3;
+	wavePos.z /= WATER_WAVE_LENGTH * 2;
```

Comment:
I prefer the noise is not stretched in one direction, but do not insist on this.

Reply:
This is so you get waves (as in a wave-front that travels). Otherwise you "wave hills" that travel, which looks less real/satisfying.

Reply:
Ah i misunderstood and thought the noise was stretched in the direction of travel. This is fine then.

Reply:
Yes the comment is confusing (and wrong actually). Hmm... What I'm really doing is increasing the frequency of the noise function in the z-dimension, so it is compressed that way and causes a wave-front.
Does that make more sense, if so I'll comment as such.

Reply:
I'd just exchange "stretched" with "compressed".

Reply:
Fine and go ahead and merge afterwards.